lib/index.js

```diff
@@ -9,7 +9,13 @@ export default class Roots {
   compile () {
     return new Promise((resolve, reject) => {
       webpack(this.config, (err, stats) => {
+        // handle all errors
+        // ref: https://webpack.github.io/docs/node.js-api.html#error-handling
         if (err) { return reject(err) }
+        let jsonStats = stats.toJson()
+        if (jsonStats.errors.length > 0) { reject(jsonStats.errors) }
+        if (jsonStats.warnings.length > 0) { reject(jsonStats.warnings) } // we may want to WARN instead of reject
```

Comment:
Yeah agreed, we'll just have it emit a warning. I'm wondering if maybe we need to convert the promise to an emitter or stream or something out of the API. I love promises but they only give us accept and reject, no progress, warning, or other stuff. It did end up being slightly an issue in the previous version.